### DISCUSSION TITLE
Utilise `app.json` when using `git:from-image`

### DISCUSSION THREAD

**Discussion by ashmckenzie** [dokku/dokku#6599] [General]
Has anyone been able to or is it even possible to have an `app.json` (https://dokku.com/docs/deployment/zero-downtime-deploys/#customizing-checks) used when deploying via `git:from-image` ?

**Comment by josegonzalez** [comment]
It should be as long as the app.json is in the workdir of the image.

**Reply by ashmckenzie** [reply]
Thanks @josegonzalez, I was hoping I could 'sideload' it or such instead of baking into in the image, but I can manage with that, thanks 🙇

**Reply by josegonzalez** [reply]
There's probably a way to do this with a custom plugin, but we otherwise extract everything needed for the deploy from the source we have.

What would be your ideal workflow here?

**Reply by ashmckenzie** [reply]
Perhaps something along the lines of `dokku --app <app> checks:set <JSON>` or `dokku --app <app> checks:upload <URL_TO_JSON>`  ?

The scenario I have is the Docker image I'm using is not mine, so I can't easily add an `app.json` to the image, unless I create a new image with an included `app.json` which isn't super difficult but it would be nice to have another option.

**Reply by josegonzalez** [reply]
What would happen if someone started including an app.json with their repo? Which takes precedence?

**Reply by ashmckenzie** [reply]
The uploaded/side loaded `app.json` I would say, with a warning advising that the checked in `app.json` was overridden ?

**Reply by josegonzalez** [reply]
So the uploaded one always overrides?

**Reply by ashmckenzie** [reply]
Thinking through the scenarios, that makes the most sense to me, yes :+1:

**Reply by josegonzalez** [reply]
What about other files? Dokku extracts others too, maybe useful there?

**Reply by ashmckenzie** [reply]
Apologies I'm not too familiar with this but I can definitely see a use case for overwriting other files inside the image.

**Reply by josegonzalez** [reply]
What about deleting the uploaded file?
